tests/virtual_machines_test.go

```diff
@@ -879,6 +886,75 @@ var _ = Describe("[rfe_id:3503][crit:medium][vendor:cnv-qe@redhat.com][level:com
 					})
 				})
 			})
+
+			Context("and creating two VMs with same mac, one on unmanaged (opted out) namespace", func() {
+				var conflictingVM *kubevirtv1.VirtualMachine
+				BeforeEach(func() {
+					macAddress := "02:00:ff:ff:ff:ff"
+
+					By(fmt.Sprintf("Adding a vm with a Mac Address %s in the managed namespace", macAddress))
+					vm1 := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("br1", macAddress)}, []kubevirtv1.Network{newNetwork("br1")})
+					err := testClient.VirtClient.Create(context.TODO(), vm1)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(net.ParseMAC(vm1.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)).ToNot(BeEmpty(), "Should successfully parse mac address")
+
+					err = addLabelsToNamespace(OtherTestNamespace, map[string]string{vmNamespaceOptInLabel: "ignore"})
+					Expect(err).ToNot(HaveOccurred(), "should be able to add the namespace labels")
+
+					By(fmt.Sprintf("Adding a vm with the same Mac Address %s on the unmanaged namespace", macAddress))
+					conflictingVM = CreateVmObject(OtherTestNamespace, false, []kubevirtv1.Interface{newInterface("br1", macAddress)}, []kubevirtv1.Network{newNetwork("br1")})
+					err = testClient.VirtClient.Create(context.TODO(), conflictingVM)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(net.ParseMAC(conflictingVM.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)).ToNot(BeEmpty(), "Should successfully parse mac address")
+				})
+
+				It("should detect duplicate macs gauge after cleaning the opt-out label and restarting kubemacpool", func() {
+					By("moving the unmanaged namespace to be managed")
+					err := cleanNamespaceLabels(OtherTestNamespace)
+					Expect(err).ToNot(HaveOccurred(), "should be able to remove the namespace labels")
+
+					By("restarting Kubemacpool to check if there are duplicates mac in the managed namespaces")
+					err = initKubemacpoolParams()
```

Comment:
This part: 
```
By("moving the unmanaged namespace to be managed")
					err := cleanNamespaceLabels(OtherTestNamespace)
					Expect(err).ToNot(HaveOccurred(), "should be able to remove the namespace labels")

					By("restarting Kubemacpool to check if there are duplicates mac in the managed namespaces")
					err = initKubemacpoolParams()
					Expect(err).ToNot(HaveOccurred())
```
should be on BeforeEach, as it should be done for this and also for the next context

Reply:
But the 2nd context (and only the 2nd) must delete the VM before the initKubemacpoolParams
so just the clean i can move to the before each if needed
make sense ?